datalad_crawler/pipelines/loris_bids_export.py

```diff
@@ -0,0 +1,138 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""A pipeline for crawling BIDS exported files from a LORIS database"""
+
+# Import necessary nodes
+from ..nodes.crawl_url import crawl_url
+from datalad.utils import updated
+from ..nodes.annex import Annexificator
+from datalad_crawler.consts import DATALAD_SPECIAL_REMOTE
+from os.path import basename, join
+
+
+import json
+# Possibly instantiate a logger if you would like to log
+# during pipeline creation
+from logging import getLogger
+lgr = getLogger("datalad.crawler.pipelines.kaggle")
+
+
+class LorisAPIBIDSExtractor(object):
+
+    def __init__(self, apibase=None, annex=None):
+        self.apibase = apibase
+        self.meta    = {}
+        self.repo    = annex.repo
```

Comment:
with the common code for `finalize`, constructor and majority of the pipeline, I think you could have avoided good amount of code duplication by establishing some `BaseLorisExtractor` to reside e.g. in `loris.py` and serve as a base class.  Then derived classes would provide only the critical difference (configuration for annex, these days also better be done via config procedures, but I guess ok as is for here/antique-crawler).

Reply:
very true! I did them one at a time when I needed them and with not much time to do proper programming ;). I'll look into it.